File: subsystem_cpp/subsystem/sdk/cqviewcontroller.cpp
#include "cqviewcontroller.hh"

struct _self_cqViewController {
    cqView::ref view;
};

cqViewController::cqViewController() {
}

void cqViewController::viewDidLoad() {
}

void cqViewController::viewDidAppear() {
}

void cqViewController::viewDidDisappear() {
}

cqView::ref cqViewController::view() {
    if (self->view == nullptr) {
        loadView();
    }
    return self->view;
}

void cqViewController::loadView() {
    self->view = cqView::create();
}

File: cusapp_lua/libluacusappscript/sdk/cqviewcontroller.lua
CQViewController = extends(CQObject, {
    view = nil,
})

function CQViewController:init()
    CQViewController.super.init(self)
end

function CQViewController:viewDidLoad()
end

function CQViewController:viewDidAppear()
end

function CQViewController:viewDidDisappear()
end

function CQViewController:view()
    if self.view == nil then
        self:loadView()
    end
    return self.view
end

function CQViewController:loadView()
    self.view = CQView:new()
end
